cpp/src/parquet/column_reader.cc

```diff
@@ -57,6 +57,11 @@
 #include "parquet/thrift_internal.h"  // IWYU pragma: keep
 #include "parquet/windows_fixup.h"    // for OPTIONAL
 
+#ifdef _MSC_VER
```

Comment:
Is it possible to move it into `parquet/windows_fixup.h`?

Reply:
I am not sure whether this is a good idea. `parquet/windows_fixup.h` is again part of the public facing API (via parquet/schema.h) so we would need to do push / pop again to avoid disabling the warning for everyone.

Reply:
I was thinking of defining some common macros for these but it seems that these patterns have already been scattered around. WDYT? @kou

Reply:
How about creating `parquet/windows_fixup_internal.h`?
Our build system doesn't install `*_internal.h`.

Reply:
I am not sure whether it is wort the effort. We are talking about two times three lines of code. Does this need a separate file?